x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/Accuracy.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe.evaluation.classification;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.Aggregations;
+import org.elasticsearch.search.aggregations.metrics.NumericMetricsAggregation;
+import org.elasticsearch.xpack.core.ml.dataframe.evaluation.EvaluationMetricResult;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * {@link Accuracy} is a metric that answers the question:
+ *   "What fraction of examples have been classified correctly by the classifier?"
+ *
+ * equation: accuracy = 1/n * Σ(y == y´)
+ */
+public class Accuracy implements ClassificationMetric {
```

Comment:
What do you think of adding a option called `classes` or `labels` or something that is a list of classes that the caller wants individual accuracy metrics for in addition to the overall accuracy?

Even for binary, knowing which label is more accurately predicted is valuable.

Reply:
I definitely agree that we should return per-label accuracy in this metric.
The question is whether we should ask the user for input (by providing an additional parameter) or we'd rather output per-label accuracy for all the labels automatically.
The response would look sth like:
```
  "overall_accuracy": 0.575,
  "actual_classes": [
    {
      "actual_class": "cat",
      "accuracy": 0.92666
    },
    {
      "actual_class": "dog",
      "accuracy": 0.25
    }
  ]
```

WDYT?

Reply:
@przemekwitek I like that format. Once we have more than two classification labels, I do think users would like to specify them. For now, we can just supply individual accuracy for all labels..

Reply:
> Once we have more than two classification labels, I do think users would like to specify them. 
Ok, let's revisit this when we go multiclass

> For now, we can just supply individual accuracy for all labels..
Done. PTAL